demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs

```diff
@@ -2,30 +2,43 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections;
+using System.Diagnostics;
 
 internal class Program
 {
     private static void Main(string[] args)
     {
-        if (args.Length == 1 && args[0] == "linux")
+        string osDetect = DetectOS();
+
+        if (args.Length >= 1)
+        {
+            osDetect = args[0];
+        }
+
+        if (osDetect == "linux")
```

Comment:
The problem with this is that the actual OS detection APIs are a query, not something that returns a value that can be switched on. And so once this sample moves to the official OS detection APIs, it will have to be rewritten back to if/else statements.